### PR TITLE
Update to julia-0.5

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
JuliaSmoothOptimizers/AmplNLReader.jl/pull/27, JuliaSmoothOptimizers/CUTEst.jl/pull/79 and JuliaSmoothOptimizers/NLPModels.jl/pull/42 have circular dependency on the tests, so it'll still show warnings.

File: src/CUTEst.jl [hunk 1]
```diff
@@ -20,13 +20,13 @@ end
 const outsdif = "OUTSDIF.d";
 const automat = "AUTOMAT.d";
 const funit   = convert(Int32, 42);
-@osx? (const linker = "gfortran") : (const linker = "ld")
-@osx? (const sh_flags = ["-dynamiclib", "-undefined", "dynamic_lookup"]) : (const sh_flags = ["-shared"]);
-@osx? (const soname = "dylib") : (const soname = "so");
+@static is_apple() ? (const linker = "gfortran") : (const linker = "ld")
+@static is_apple() ? (const sh_flags = ["-dynamiclib", "-undefined", "dynamic_lookup"]) : (const sh_flags = ["-shared"]);
+@static is_apple() ? (const soname = "dylib") : (const soname = "so");
```
Comment (dpo) [inline src/CUTEst.jl]: Don't we need `Compat.String` here?

File: src/CUTEst.jl [hunk 2]
```diff
@@ -20,13 +20,13 @@ end
...
 
 type CUTEstException <: Exception
   info :: Int32
-  msg  :: ASCIIString
+  msg  :: String
```
Comment (dpo) [inline src/CUTEst.jl]: I meant here...

Comment (abelsiqueira) [inline src/CUTEst.jl]: Imported `Compat.String`.

**Comment by abelsiqueira** [comment]
Travis fixed. This is ready.

File: test/consistency.jl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+path = joinpath(Pkg.dir("NLPModels"), "test")
+include(joinpath(path, "consistency.jl"))
+
+problems = [:brownden, :hs5, :hs6, :hs10, :hs11, :hs14]
+for problem in problems
+  problem_s = string(problem)
+  include(joinpath(path, "$problem_s.jl"))
+
+  problem_f = eval(problem)
+  nlp_jump = JuMPNLPModel(problem_f())
```
Comment (dpo) [inline test/consistency.jl]: This is `MathProgNLPModel` now.

Comment (abelsiqueira) [inline test/consistency.jl]: Fixed

**Comment by abelsiqueira** [comment]
Ready

**Pull request merged**